### ISSUE TITLE
Questions Regarding Reproducing Your Paper

### ISSUE THREAD

**Issue by Vo-Linh** [wangshanwen001/RS-MUCA#2]
Dear @wangshanwen001 

I am attempting to reproduce your paper. However, I have encountered some issues and would appreciate your insights.

1. I noticed that some file names in your dataset split do not match those in the public LoveDA dataset. Did you also encounter this issue?

2. Due to this mismatch, I decided to run the experiments using the original LoveDA dataset with its default train/val/test split. However, the model’s performance was significantly lower than expected. I have attached an image illustrating the results.

3. During my experiments, I encountered the following warning:
`warnings.warn(f"input's size at dim={feature_dim} does not match num_features. ")  `
Did you also face this issue when running your experiments?

4. Lastly, please provide your best model so that I can evaluate it using a standard dataset.

Thank you for your time and support. I look forward to hearing back from you.

![Image](https://github.com/user-attachments/assets/c3b1f42b-a84a-4b6c-a509-0b4f3f07df53)

![Image](https://github.com/user-attachments/assets/df53a575-89dd-4102-96b6-14d29c79398f)

**Comment by wangshanwen001** [comment]
I will address your questions point by point:
1.As stated in our paper, we cropped images from both Potsdam and LoveDA datasets to 512×512 pixels. This effectively reduces GPU memory consumption since the original Potsdam images (6000×6000 pixels) exceed our hardware training capabilities. For LoveDA dataset, we maintained the official naming convention (1, 2, 3...) for cropped images, but there are more images cropped than in the original dataset. This explains why some indices in our split files don't match the original dataset. It's worth noting that image cropping for efficient training is a common practice in the field, and our approach follows previous research methodologies.

2.Even when using the standard LoveDA dataset, such significant performance degradation shouldn't occur. I noticed substantial fluctuations in your early-stage training results (as shown in your provided images), which might stem from not loading Segformer's pretrained weights - this omission could indeed impair training effectiveness. Additionally, since I'm unaware of your train/val/test split ratios, please note we used a 6:2:2 ratio as specified in our paper.

3.We didn't encounter this warning during our training. The input_shape variable in train.py was set to our cropped size (512×512). Using original image dimensions might trigger this warning and potentially degrade training performance.

4.While I'm uploading the best [LoveDA.pth](https://pan.baidu.com/s/18i2arPvZD3tKRa3ZcjaKgg) file here. Code = ``c9vq``. We cannot guarantee identical results if your dataset configuration differs from our paper specifications. The attached image shows our training loss reduction curve for your reference.

![Image](https://github.com/user-attachments/assets/4d9b2fe7-dd11-402e-ae48-a4205efca1ec)
